addons/social_share/models/snshare_campaign_element.py

```diff
@@ -0,0 +1,73 @@
+import base64
+
+from odoo import _, api, exceptions, fields, models
+
+class CampaignElement(models.Model):
+    _name = 'snshare.campaign.element'
+    _description = 'Social Share Campaign Element'
+    _sql_constraints = [('role_uniq', "unique(campaign_id, role)", "Each campaign should only have one element for each role.")]
```

Comment:
:socks: we generally define those after fields (fields, constraints, compute, crud, ...)